src/span.js

```diff
@@ -189,81 +254,60 @@ export default class Span {
-     * opentracing integration (e.g., arbitrary application data from a web
-     * client can make it, transparently, all the way into the depths of a
-     * storage system), and with it some powerful costs: use this feature with
-     * care.
-     *
-     * IMPORTANT NOTE #1: setBaggageItem() will only propagate baggage items to
-     * *future* causal descendants of the associated Span.
-     *
-     * IMPORTANT NOTE #2: Use this thoughtfully and with care. Every key and
-     * value is copied into every local *and remote* child of the associated
-     * Span, and that can add up to a lot of network and cpu overhead.
-     *
-     * @param {string} key
-     * @param {string} value
-     */
-    setBaggageItem(key, value) {
-        if (process.env.NODE_ENV === 'debug') {
-            if (arguments.length !== 2) {
-                throw new Error('Invalid number of arguments');
-            }
-        }
-        if (this._imp) {
-            this._imp.setBaggageItem(key, value);
-        }
+        // Do not return this. The Span generally should not be used after it
+        // is finished so chaining is not desired in this context.
     }
 
-    /**
-     * Returns the value for a baggage item given its key.
-     *
-     * @param  {string} key
-     *         The key for the given trace attribute.
-     * @return {string}
-     *         String value for the given key, or undefined if the key does not
-     *         correspond to a set trace attribute.
-     */
-    getBaggageItem(key) {
-        if (process.env.NODE_ENV === 'debug') {
-            if (arguments.length !== 1) {
-                throw new Error('Invalid number of arguments');
-            }
-        }
-        if (this._imp) {
-            return this._imp.getBaggageItem(key);
-        }
-        return undefined;
-    }
 
     // ---------------------------------------------------------------------- //
-    // Private and non-standard methods
+    // Methods to be implemented by derived classes
     // ---------------------------------------------------------------------- //
 
-    /**
-     * Constructs a new Span object, this method should not be called directly;
-     * Tracer.startSpan() or Tracer.join() should be used instead.
-     */
-    constructor(imp) {
-        this._imp = imp;
+    // By default returns a no-op SpanContext.
+    _context() {
+        return new SpanContext();
     }
 
-    /**
-     * Returns the Span implementation object. The returned object is by its
-     * nature entirely implementation-dependent.
-     */
-    imp() {
-        return this._imp;
+    // By default returns a no-op tracer.
+    //
+    // The base class could store the tracer that created it, but it does not
+    // in order to ensure the no-op span implementation has zero members,
+    // which allows V8 to aggressively optimize calls to such objects.
+    _tracer() {
+        return noopTracer;
+    }
+
+    // By default does nothing
+    _setOperationName(name) {
```

Comment:
Yes, the intention is that derived classes would implement these methods, per line 265.